### ISSUE TITLE
Add loading option

### ISSUE THREAD

**Issue by ricardopacheco** [flyerhq/react-native-link-preview#17]
It`s possible to add a component loading when loading server data?

**Comment by demchenkoalex** [comment]
Hi @ricardopacheco it is out of scope for this library. It can be presented in many ways that's why we decided to not include it at all. Plus, if you have loading indicator user will expect something and if the link preview failed you will ideally need to include error handling too, which complicates things a lot. Anyway, we provide render props for everything so you can place the activity indicator where you want, we only lack a onError callback which you will need to use to set loading to false. We can do it I guess.

**Comment by ricardopacheco** [comment]
I understand @demchenkoalex  but in this case I could just add the events for the users to handle as they wish, I believe that would make the tool interesting for top of the line.

My suggestions: onError, onSuccess.

One question: In the current version, how do I render the component only after it has been completed?

**Comment by demchenkoalex** [comment]
you already have `onSuccess` in form of `onPreviewDataFetched`.

Again, `onPreviewDataFetched` marks a completed preview. But in the current version (I'm not sure) it might fire even if there was an error getting the preview. We should separate that to the `onError` callback

**Comment by ricardopacheco** [comment]
@demchenkoalex Could you give me an example of implementation?

**Comment by demchenkoalex** [comment]
import { LinkPreview, PreviewData } from '@flyerhq/react-native-link-preview'

const [previewData, setPreviewData] = useState<PreviewData | undefined>()

const handlePreviewDataFetched = (data: PreviewData) => {
  setPreviewData(data)
  // Do whatever additional logic you need to run on success
}

<LinkPreview
  onPreviewDataFetched={handlePreviewDataFetched}
  previewData={previewData}
  text='https://apple.com'
/>

**Comment by ricardopacheco** [comment]
In my case, I am using a rich editor (which detects when a link is inserted). I always capture the first link of the publisher's content. When the link is detected, I change a state called `linkPreview` with the value of the URL via useEffect. In rendering, it detects and renders. An outline:

```
const [editorState, setEditorState] = useState ('');
const [linkPreview, setLinkPreview] = useState (null);
...
useEffect(() => {
  if (editorState.length > 0) {
    setLinkPreview('https://youtube.com/');
  } else {
    setLinkPreview(null);
  }
}, [editorState]);
...

return (
  <View style = {styles.postEditorPreview}>
    {linkPreview && (
      <View style = {styles.postEditorPreviewLink}>
        <LinkPreview text = {linkPreview} renderText = {() => null} />
      </View>
)}
```

This code worked as expected, since it is only actually rendering the preview, and not the text + preview as i wish.

**Comment by demchenkoalex** [comment]
Hi @ricardopacheco so I went though the library once more and here are the answers:

1. Loading state - you start by having it set to true whenever you render a preview data (or re-render). When `onPreviewDataFetched` callback is triggered that means the library did everything and you can set loading to false. Here you have your loading state.
2. Error callback - you can use the same `onPreviewDataFetched` callback and check the preview data - if it has nothing or only link inside, that means there was an error or the page is totally empty, which is sort of an error too. So you have your error state here.
3. You mentioned that `<LinkPreview text={linkPreview} onPreviewDataFetched={() => console.log('Done')} />` `console.log` is not triggered, but I tried the example project and checked the code and it will be triggered for everything, including empty line. You must have had a conditional render or something since you didn't see the `console.log`, which can be fixed on your side.
